compiler/rustc_const_eval/src/transform/validate.rs

```diff
@@ -261,7 +262,15 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                     )
                 }
             };
-            match parent_ty.ty.kind() {
+
+            let kind = match parent_ty.ty.kind() {
+                ty::Opaque(def_id, substs) => {
+                    self.tcx.type_of(def_id).subst(self.tcx, substs).kind()
```

Comment:
that could cause query cycle if it were called before `mir_borrowck` afaict :thinking: 

It's surprising to me that this doesn't already happen :sweat_smile: Do you know why this is fine?

Reply:
@oli-obk do you have an opinion about that?